service/matching/matchingEngine.go

```diff
@@ -275,7 +275,9 @@ func (e *matchingEngineImpl) AddWorkflowTask(
 		return false, err
 	}
 
-	taskQueue, err := e.redirectToVersionedQueueForAdd(hCtx, origTaskQueue, addRequest.WorkerVersionStamp, taskQueueKind)
+	isFirstWorkflowTask := addRequest.ScheduledEventId == common.FirstWorkflowTaskScheduledEventID
```

Comment:
This is kind of messy, but the alternative seemed worse:
I tried adding a flag in AddWorkflowTaskRequest to keep this logic in history, but matching also needs to construct AddWorkflowTaskRequests when forwarding, including spooled tasks, and then that means we need to add the flag to the persisted matching task also. That seems a little overly complicated for no real benefit. But possible if you think this coupling is really that bad

Reply:
switched to passing everything through. also fixed forwarding